src/main/java/ch/cern/eam/wshub/core/services/workorders/ChecklistService.java

```diff
@@ -26,6 +26,6 @@ WorkOrderActivityChecklistSignatureResponse eSignWorkOrderActivityChecklist(Info
             throws InforException;
 
     @Operation(logOperation = INFOR_OPERATION.WO_CHECKL_ESGN_R)
-    WorkOrderActivityChecklistSignatureResult[] getSignatures(InforContext context, String workOrderCode, String activityCode)
+    WorkOrderActivityChecklistSignatureResult[] getSignatures(InforContext context, String workOrderCode, String activityCode, TaskPlan taskPlan)
```

Comment:
Is it possible to read the task plan when the service is called, so that the caller does not need to do that themselves? Would there be any disadvantages to that?

Reply:
In LaborBookingServiceImpl the Taskplan is later used to determine if the signatures should be displayed at all, so this allows to minimise the number of calls to the getTaskplan WS. However, the I can refactor LaborBookingServiceImpl and move the population of the checklist field and the signature field to ChecklistService in a new method and hide this method by making it private. The new population method will only need the activity as a parameter.

Reply:
Resolved via Zoom; keeping as is.